py/sentry_streams/user_functions/function_template.py

```diff
@@ -0,0 +1,79 @@
+from abc import ABC, abstractmethod
+from typing import Any, Generic, Optional, TypeVar
+
+InputType = TypeVar("InputType")
+OutputType = TypeVar("OutputType")
+IntermediateType = TypeVar("IntermediateType")
+
+
+class AggregationBackend(ABC, Generic[OutputType]):
+    """
+    A storage backend that is meant to store windowed aggregates. Configurable
+    to the type of storage.
+    """
+
+    @abstractmethod
+    def flush_aggregate(self, aggregate: OutputType) -> None:
+        """
+        Flush a windowed aggregate to storage. Takes in the output from
+        the Accumulator.
+        """
+
+
+class Accumulator(ABC, Generic[InputType, IntermediateType, OutputType]):
+    """
+    The standard Accumulator template.
+    Define these functions to build a custom
+    Accumulator for aggregation.
+    """
+
+    def __init__(self, backend: Optional[AggregationBackend[OutputType]] = None):
+        self.backend = backend
+
+    @abstractmethod
+    def create(self) -> IntermediateType:
+        """
+        Initialize a new Accumulator with seed values.
+        """
+        raise NotImplementedError
+
+    @abstractmethod
+    def add(self, acc: IntermediateType, value: InputType) -> IntermediateType:
+        """
+        Add values to the Accumulator. Can produce a new type which is different
+        from the input type.
+        """
+        raise NotImplementedError
+
+    @abstractmethod
+    def get_output(self, acc: IntermediateType) -> OutputType:
```

Comment:
What about `get_value` ? 
It seems more in line with the semantics of an accumulator which is a system that accumulates data into a state and that can return the state at any moment. 
get_output seems to imply there is some sort of computation done by the accumulator, which may not be the case.

This may sound pedantic but I believe that, when defining an api, naming things in a way that precisely identify the semantics pays back later when people use it.